### PR TITLE
Add quoteCart subscription

### PR DESCRIPTION
**Pull request by tobiasbexelius**
# Jira Issue: [MX-378]

## What?
- A quoteCart subscription
- Related PRs: https://github.com/HedvigInsurance/member-service/pull/588, https://github.com/HedvigInsurance/underwriter/pull/421

## Why?
- Because it's nice :)

## Optional checklist
- [ ] Unit tests written
- [x] Tested locally
- [x] Codescouted

[MX-378]: https://hedvig.atlassian.net/browse/MX-378?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by robinandeer** [comment]
Cool stuff! So we could use this when waiting for signing status I suppose?

**Comment by tobiasbexelius** [comment]
Yes. ATM signing updates is the only thing that triggers updates though, not e.g. the mutations, which you shouldn't need because it returns the updated object, but it may be a bit confusing... I will look in to fixing that

File: src/resolvers/quote-cart/quoteCart.ts [hunk 1]
```diff
@@ -338,3 +359,44 @@ const transformSessionDto = (dto: OnboardingSessionDto): QuoteCart => {
...
 }
+
+export const withTransform = <A, B>(
+  asyncIterator: AsyncIterator<{ id: A }>,
+  transform: () => Promise<B>,
+): AsyncIterator<{ quoteCart: B }> => {
+  return {
+    next() {
+      return asyncIterator.next().then(async ({ value, done }) => {
+        console.log(`####### THE VALUE: ${value}, done ${done}`)
```
Comment (FredrikAreschoug) [inline src/resolvers/quote-cart/quoteCart.ts:370]: Remove right?

File: src/resolvers/quote-cart/quoteCart.ts [hunk 2]
```diff
@@ -190,6 +193,24 @@ export const QuoteCartSubset: ResolverSubset = {
...
+  Subscription: {
+    quoteCart: {
+      subscribe: async (
+        _parent,
+        input,
+        { upstream, pubsub },
+      ): Promise<AsyncIterator<{ quoteCart: QuoteCart }>> => {
+        // throws if cart does not exist
+        await upstream.underwriter.getOnboardingSession(input.id)
+        await pubsub.publish(`SIGN_EVENTS.${input.id}`, `init`)
```
Comment (FredrikAreschoug) [inline src/resolvers/quote-cart/quoteCart.ts:205]: Should we maybe separate this from the sign events that already exists? Call them something else?

**Review by FredrikAreschoug — APPROVED** [review]
Looks good added minor comments, but for me it can go out :slightly_smiling_face:

File: src/resolvers/quote-cart/quoteCart.ts (hunk 2 above)
Comment (tobiasbexelius) [reply src/resolvers/quote-cart/quoteCart.ts:205]: Maybe, the old ones will be removed soonish though

File: src/resolvers/quote-cart/quoteCart.ts (hunk 1 above)
Comment (tobiasbexelius) [reply src/resolvers/quote-cart/quoteCart.ts:370]: oops!

**Pull request merged**